app/backend/hiStoryBackend/api/views/auth/views_v1.py

```diff
@@ -1,12 +1,35 @@
 from django.http import JsonResponse
+from django.contrib.auth.hashers import check_password, make_password
+import json
+from api.models import User
+import jwt
+from hiStoryBackend.settings import SECRET_KEY
 from api.helpers import json_response_helper as jrh
 
-
 def sign_in(request):
-    if not request.method == 'POST':
-        return jrh.bad_request(['Unsupported HTTP verb'])
+	if not request.method == 'POST':
+		return jrh.bad_request(['Unsupported HTTP verb'])
+
+	content = json.loads(request.body)
+	if 'username' in content:
+		try:
+			user = User.objects.get(username = content['username'])
+		except User.DoesNotExist:
+			return jrh.unauthorized(["username does not exist"])
+	elif 'email' in content:
+		try:
+			user = User.objects.get(email = content['email'])
+		except User.DoesNotExist:
+			return jrh.unauthorized(["email does not exist"])
+	else:
+		return jrh.bad_request(["username and email are both missing"])
+
+	if check_password(content['password'], user.password_hash) == False:
```

Comment:
We should check for the presence of `password` in the request, and if not present, we should return `jhr.bad_request()`.

Also, you can use `if not check_password(...):`